cmd/account/o-auth-enrollment/overrides.go

```diff
@@ -0,0 +1,104 @@
+package o_auth_enrollment
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/databricks/cli/libs/cmdio"
+	"github.com/databricks/databricks-sdk-go"
+	"github.com/databricks/databricks-sdk-go/retries"
+	"github.com/databricks/databricks-sdk-go/service/oauth2"
+	"github.com/spf13/cobra"
+)
+
+func promptForBasicAccountConfig(ctx context.Context) (*databricks.Config, error) {
+	if !cmdio.IsInTTY(ctx) {
+		return nil, fmt.Errorf("this command requires a TTY")
+	}
+	// OAuth Enrollment only works on AWS
+	host, err := cmdio.DefaultPrompt(ctx, "Host", "https://accounts.cloud.databricks.com")
+	if err != nil {
+		return nil, fmt.Errorf("host: %w", err)
+	}
+	accountID, err := cmdio.SimplePrompt(ctx, "Account ID")
+	if err != nil {
+		return nil, fmt.Errorf("account: %w", err)
+	}
+	username, err := cmdio.SimplePrompt(ctx, "Username")
+	if err != nil {
+		return nil, fmt.Errorf("username: %w", err)
+	}
+	password, err := cmdio.Secret(ctx, "Password")
+	if err != nil {
+		return nil, fmt.Errorf("password: %w", err)
+	}
+	return &databricks.Config{
+		Host:      host,
+		AccountID: accountID,
+		Username:  username,
+		Password:  password,
+	}, nil
+}
+
+func enableOAuthForAccount(ctx context.Context, cfg *databricks.Config) error {
+	ac, err := databricks.NewAccountClient(cfg)
+	if err != nil {
+		return fmt.Errorf("account client: %w", err)
+	}
+	// The enrollment is executed asynchronously, so the API returns HTTP 204 immediately
+	err = ac.OAuthEnrollment.Create(ctx, oauth2.CreateOAuthEnrollment{
+		EnableAllPublishedApps: true,
+	})
+	if err != nil {
+		return fmt.Errorf("enroll: %w", err)
+	}
+	enableSpinner := cmdio.Spinner(ctx)
+	// The actual enrollment take a few minutes
```

Comment:
if it takes that longs, better show some notice for the customers that operation can take a while

Reply:
isn't spinner doing that?